### PR TITLE
Array trait impl comment/doc fixes

### PR DESCRIPTION
**Pull request by est31**
Two small doc/comment fixes regarding trait implementations on arrays.

### CODE REVIEW

**Comment by Dylan-DPC** [comment]
failed spuriously

@bors retry

File: library/core/src/array/mod.rs
Comment (camelid) [inline library/core/src/array/mod.rs:356]:
```suggestion
// The Default impls cannot be done with const generics because `[T; 0]` doesn't
```

**Review by camelid — CHANGES_REQUESTED** [review]

**Comment by camelid** [comment]
Another spurious network failure :/

**Comment by est31** [comment]
@camelid will the PR be reapproved if I change it after it has been approved? I've done that with my other PR and there is no reaction since: #78465

**Comment by est31** [comment]
Shrug, I'll just try it. r? @camelid

**Comment by camelid** [comment]
> @camelid will the PR be reapproved if I change it after it has been approved? I've done that with my other PR and there is no reaction since: #78465

@est31 No, bors will not merge it if it has been changed since being approved. I think it's a security feature.

(Btw, I don't have r+ privileges, so unfortunately I wouldn't have been able to approve this.)

**Comment by Dylan-DPC** [comment]
don't worry i have reapproved it, so it is in the queue

**Comment by camelid** [comment]
Yeah, I just re-assigned you so that if there are issues you can re-approve it :)

**Pull request merged**
